src/components/StatesButton/StatesButton.tsx

```diff
@@ -44,18 +62,56 @@ export class StatesButton extends React.Component<
     );
   }
 
+  private renderContent(): React.ReactElement | string {
+    const {
+      state,
+      idleContent,
+      inProgressContent,
+      failureContent,
+      successContent,
+    } = this.props;
+
+    switch (state) {
+      case BUTTON_STATES.IDLE:
+        return idleContent;
+      case BUTTON_STATES.IN_PROGRESS:
+        return inProgressContent;
+      case BUTTON_STATES.FAILURE:
+        return failureContent;
+      case BUTTON_STATES.SUCCESS:
+        return successContent ? successContent : this.renderCheck();
+      default:
+        return idleContent;
+    }
+  }
```

Comment:
According to the guidelines we got from neil:
https://docs.google.com/document/d/19BWUuPxIUQ4CLKL6KBG6KRsPmLutq_6RFtYimxWZ4kE/edit?usp=sharing

The different content types should be rendered with different tags.
`idle` and `loading` are buttons, but the `success` and `error` should just be divs (absolutely positioned).
While in the `success` and `error` states, the `div`'s and `button` should have a`tabindex=-1`.
We might probably want to add an `aria-label` to the `div`'s in case the content is of type `string`.